src/components/Dropdown/Dropdown.tsx

```diff
@@ -305,6 +325,44 @@ export const Dropdown: FC<DropdownProps> = React.memo(
         }
       };
 
+      const handleDropdownClick = (event: React.MouseEvent): void => {
+        // Check if the click is on an element inside a list item
+        const isElementInsideListItem = (target: EventTarget): boolean => {
+          const element = target as HTMLElement;
+          // Walk up the DOM tree to find if we're inside a list item
+          let currentElement = element;
+          while (
+            currentElement &&
+            !currentElement.matches('[role="option"], .list-item, li')
+          ) {
+            // If we reach the dropdown wrapper, we're not inside a list item
+            if (currentElement.matches(`.${styles.dropdownWrapper}`)) {
+              return false;
+            }
+            currentElement = currentElement.parentElement;
+          }
+
+          // If we found a list item, check if the clicked element is a child of it
+          // but not the list item itself
+          if (currentElement) {
+            return element !== currentElement;
+          }
+
+          return false;
+        };
+        // If clicking on an element inside a list item and closeOnElementClick is false,
+        // don't close the dropdown
+        if (!closeOnElementClick && isElementInsideListItem(event.target)) {
```

Comment:
you might need to check multiple as well.. If you have multi selection, then the close on element click might not help